entity generator - String field RegExp pattern not escaped in Java

Generated an entity with a string field validated with this regular expression pattern:
^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$
to validate IPv4 addresses.

Compiling the application fails with:
```java
MyEntity.java:32: error: illegal escape character
    @Pattern(regexp = "^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$")
```
(It doesn't like that `\.` in the middle there.

MyEntity.java, should be generated as:
```java
    @NotNull
    @Pattern(regexp = "^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$")
    @Field("ip")
    private String ip;
```
(With the escaped `\\.`)

The javascript side functions fine. Its a minor issue that is easily corrected after entity generation but does cause compile time error.

JHipster v2.17.0

--- a/entity/index.js
+++ b/entity/index.js
@@ -562,11 +562,12 @@ EntityGenerator.prototype.askForFields = function askForFields() {
                 fieldValidateRulesMinlength: props.fieldValidateRulesMinlength,
                 fieldValidateRulesMaxlength: props.fieldValidateRulesMaxlength,
                 fieldValidateRulesPattern: props.fieldValidateRulesPattern,
+                fieldValidateRulesPatternJava: props.fieldValidateRulesPattern.replace(/\\/g, '\\\\'),
                 fieldValidateRulesMin: props.fieldValidateRulesMin,
                 fieldValidateRulesMax: props.fieldValidateRulesMax,
                 fieldValidateRulesMinbytes: props.fieldValidateRulesMinbytes,
                 fieldValidateRulesMaxbytes: props.fieldValidateRulesMaxbytes
-                }
+                };
 
             fieldNamesUnderscored.push(_s.underscored(props.fieldName));
             this.fields.push(field);

--- a/entity/templates/src/main/java/package/domain/field_validators.ejs
+++ b/entity/templates/src/main/java/package/domain/field_validators.ejs
@@ -35,7 +35,7 @@ if (field.fieldValidate == true) {
         validators.push('@Max(value = ' + field.fieldValidateRulesMax + ')');
     }
     if (rules.indexOf('pattern') != -1) {
-        validators.push('@Pattern(regexp = "' + field.fieldValidateRulesPattern  + '")');
+        validators.push('@Pattern(regexp = "' + field.fieldValidateRulesPatternJava  + '")');
     }
     result =  validators.join('\n    ');
 } -%>

--- a/entity/templates/src/main/java/package/web/rest/dto/_EntityDTO.java
+++ b/entity/templates/src/main/java/package/web/rest/dto/_EntityDTO.java
@@ -37,7 +37,7 @@ public class <%= entityClass %>DTO implements Serializable {
     @Size(min = <%= fields[fieldId].fieldValidateRulesMinbytes %>, max = <%= fields[fieldId].fieldValidateRulesMaxbytes %>)<% } %><% if (fields[fieldId].fieldValidateRules.indexOf('min') != -1) { %>
     @Min(value = <%= fields[fieldId].fieldValidateRulesMin %>)<% } %><% if (fields[fieldId].fieldValidateRules.indexOf('max') != -1) { %>
     @Max(value = <%= fields[fieldId].fieldValidateRulesMax %>)<% } %><% if (fields[fieldId].fieldValidateRules.indexOf('pattern') != -1) { %>
-    @Pattern(regexp = "<%= fields[fieldId].fieldValidateRulesPattern %>")<% } } %><% if (fields[fieldId].fieldType == 'byte[]') { %>
+    @Pattern(regexp = "<%= fields[fieldId].fieldValidateRulesPatternJava %>")<% } } %><% if (fields[fieldId].fieldType == 'byte[]') { %>
     @Lob<% } %>
     private <%= fields[fieldId].fieldType %> <%= fields[fieldId].fieldName %>;<% } %><% for (relationshipId in relationships) {
     otherEntityRelationshipName = relationships[relationshipId].otherEntityRelationshipName;%><% if (relationships[relationshipId].relationshipType == 'many-to-many' && relationships[relationshipId].ownerSide == true) { %>

